src/main/java/com/greenfoxacademy/goddesstribesbackend/models/entities/Soldier.java

```diff
@@ -0,0 +1,86 @@
+package com.greenfoxacademy.goddesstribesbackend.models.entities;
+
+import javax.persistence.*;
+import java.time.LocalDateTime;
+
+@Entity
+public class Soldier {
+
+  public static final int CREATION_COST = 10;
+  public static final int UPGRADING_COST = 5;
+  public static final int NEEDED_TIME = 1;
+  private static final int START_LEVEL = 1;
+  private static final int START_CONSUMPTION_RATE = 1;
+
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  private Long id;
+
+  private int level;
+  private int consumptionRate;
+  private LocalDateTime startedAt;
+  private LocalDateTime finishedAt;
+
+  @ManyToOne(optional = false)
+  private Barrack barrack;
+
+  public Soldier() {
+  }
+
+  public Soldier(Barrack barrack) {
+    level = START_LEVEL;
+    consumptionRate = START_CONSUMPTION_RATE;
+    startedAt = LocalDateTime.now();
+    finishedAt = startedAt.plusMinutes(NEEDED_TIME);
+    this.barrack = barrack;
```

Comment:
Why do we store this here? Does the Soldier have to know which Barrack is he in?